pkg/gitutils/gitutils.go

```diff
@@ -30,19 +31,45 @@ func Clone(remoteURL string) (string, error) {
 		return "", err
 	}
 
-	fragment := u.Fragment
-	clone := cloneArgs(u, root)
+	if out, err := gitWithinDir(root, "init"); err != nil {
+		return "", errors.Wrapf(err, "failed to init repo at %s: %s", root, out)
+	}
+
+	ref, subdir := getRefAndSubdir(u.Fragment)
+	fetch := fetchArgs(u, ref)
 
-	if output, err := git(clone...); err != nil {
-		return "", fmt.Errorf("Error trying to use git: %s (%s)", err, output)
+	if u.Fragment != "" {
+		u.Fragment = ""
```

Comment:
nit: probably more clear without the if, since it ends up being the same value in all cases anyway. 

```go
u.Fragment = ""
```